pkg/il/function.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package il
+
+// Function contains metadata about an IL-based function that is implemented in a program.
+type Function struct {
+
+	// Parameters is the type of the input parameters to the function.
+	Parameters []Type
+
+	// ReturnType is the return type of the function.
+	ReturnType Type
+
+	// ID is the id of the function. It is also the id of the name of the function in the strings table.
+	ID uint32
+
+	// Address is the address of the first opcode for the function within the bytecode.
+	Address uint32
+
+	// Length is the length of the function, in uint32s, within the bytecode.
+	Length uint32
+}
+
+// FunctionsTable contains a set of functions, organized by their ids.
+type FunctionsTable struct {
```

Comment:
I think both StringsTable and FunctionsTable need to drop the plural: FunctionTable and StringTable.